### PR TITLE
fix: Fix virtual machine warnings for max price of the billing profile and the type of the identity - `avm/res/compute/virtual-machine`

### PR DESCRIPTION
**Pull request by pmalarme**
## Description

This PR fixes 2 Bicep warning BCP036:
* Missing space between `SystemAssigned` and `UserAssigned` for the type of the identity
* The value of `maxPrice` for the billing profile needs to be an integer or a json decimal: https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2022-11-01/virtualmachines?pivots=deployment-language-bicep#billingprofile. By using `json()` function, I ensure that this value is converted to `int` or json decimal.

## Pipeline Reference

| Pipeline |
| -------- |
|          |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/compute/virtual-machine/main.bicep [hunk 1]
```diff
@@ -330,7 +330,7 @@ var formattedUserAssignedIdentities = reduce(
 var identity = !empty(managedIdentities)
   ? {
       type: (extensionAadJoinConfig.enabled ? true : (managedIdentities.?systemAssigned ?? false))
-        ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned')
+        ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned, UserAssigned' : 'SystemAssigned')
```
Comment (rahalan) [inline avm/res/compute/virtual-machine/main.bicep:333]: Why add the additional whitespace? Currently the same code is used multiple times in other modules as well.

**Comment by rahalan** [comment]
@pmalarme good catch! Thanks for that. We should add a test for the billing profile. Do you want to add that, e.g. to the Windows.Max test? Also make sure to regenerate the Readme after that and run the workflow. Thanks!

File: avm/res/compute/virtual-machine/main.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/compute/virtual-machine/main.bicep:333]: I think this comes down to slight differences in the modules. Some show warnigs if its not 1:1 what his particular RP expects. @pmalarme, I presume this is why you added the whitespace?

Comment (pmalarme) [reply avm/res/compute/virtual-machine/main.bicep:333]: Yes it is why

**Comment by rahalan** [comment]
@pmalarme I will include your changes into the PR #1515

**Review by rahalan — APPROVED** [review]

**Pull request merged**